src/main/java/net/apnic/whowas/ip/controller/IpHistoryRouteController.java

```diff
@@ -57,7 +69,18 @@ public IpHistoryRouteController(RDAPControllerUtil rdapControllerUtil)
             throw new MalformedRequestException(ex);
         }
 
-        return rdapControllerUtil.historyResponse(request, range);
+        int pfxCap = range.prefixSize() +
+            (range.low().getAddressFamily() == IP.AddressFamily.IPv4 ? 8 : 16);
+
+        List<ObjectHistory> ipHistory =
+            historyTree
+                .intersecting(range)
+                .filter(t -> t.first().prefixSize() <= pfxCap)
```

Comment:
Does this count as truncating the response? (Meaning the truncate notices need to be applied)

Reply:
No

Reply:
What about sections 3.1 + 4 in the RFC draft?

> **3.1. IP Network History Path Segment Specification**
> ... an historical
> query selects all networks intersecting the query range.
>
> A server MAY choose to limit the number of networks selected by a
> query, by limiting the size of sub-networks, or the total number, or
> some other means.

> **4. Query Processing**
> ... A server MAY choose to limit the responses given to a history query.
> ... If a server limits responses in this way, it SHOULD indicate to the user
> that it has done so ...

Reply:
Correct my apologies. Has been in there since day dot. A seperate issue will need to be created for this.